src/js/actions.js

```diff
@@ -183,27 +185,44 @@ function defaultParams(category, index) {
 }
 
 export function indexNav(path, category, query) {
-  history.pushState(null, (path || '/' + category) + '?q=' + encodeURIComponent(query.fullText));
-  //history.pushState(null, '/' + category, {q: query.fullText});
-  return { type: INDEX_NAV, category: category, query: query };
+  const queryKey = Object.keys(query);
```

Comment:
This feels a bit too much like not treating the query as an opaque object. In another project, we have another way of handling this aspect. I'll send you a direct message with details.